### PR TITLE
circleci: add centos6 as a new platform fo building ctags executable

### PR DESCRIPTION
**Pull request by masatake**
Signed-off-by: Masatake YAMATO <yamato@redhat.com>

### CODE REVIEW

**Comment by masatake** [comment]
This can be merged after merging #2798.

File: circle.yml [hunk 1]
```diff
@@ -225,6 +225,41 @@ jobs:
...
+       - run:
+           name: Rearrange CentOS-Base.repo file
+           command: |
+             sed -i "s|#baseurl=|baseurl=|g" /etc/yum.repos.d/CentOS-Base.repo
+             sed -i "s|mirrorlist=|#mirrorlist=|g" /etc/yum.repos.d/CentOS-Base.repo
+             sed -i "s|http://mirror\.centos\.org/centos/\$releasever|https://vault\.centos\.org/6.10|g" /etc/yum.repos.d/CentOS-Base.repo
+       - run:
+           name: Install tools for buildig
+           command: |
+             yum install -y git gcc make autconf automake pkgconfig
```
Comment (leleliu008) [inline circle.yml:242]: `autconf ` spelling mistake. Actually, `autoconf` will installed along with `automake`

Comment (masatake) [reply circle.yml:242]: Thank you!

File: circle.yml [hunk 2]
```diff
@@ -225,6 +225,38 @@ jobs:
...
+           name: Build
+           command: |
+             (
+             cd ctags
+             bash ./autogen.sh
+             ./configure --enable-debugging
+             make -j 2
+             )
+       - run:
+           name: Don't run test cases because python3 is no available
```
Comment (leleliu008) [inline circle.yml:257]: Red Hat has added `Python 3.4` for `CentOS 6` through the `EPEL` repository.

easiest way to install `Python 3.4` on `CentOS 6` from `EPEL` repository:

```
yum install -y epel-release
yum install -y python34
```

File: circle.yml [hunk 3]
```diff
@@ -225,6 +225,38 @@ jobs:
...
+       - image: centos:6
+     steps:
+       - run:
+           name: Rearrange CentOS-Base.repo file
+           command: |
+             sed -i "s|#baseurl=|baseurl=|g" /etc/yum.repos.d/CentOS-Base.repo
+             sed -i "s|mirrorlist=|#mirrorlist=|g" /etc/yum.repos.d/CentOS-Base.repo
+             sed -i "s|http://mirror\.centos\.org/centos/\$releasever|https://vault\.centos\.org/6.10|g" /etc/yum.repos.d/CentOS-Base.repo
+       - run:
+           name: Install tools for buildig
```
Comment (leleliu008) [inline circle.yml:240]: `buildig ` spelling mistake

File: circle.yml [hunk 4]
```diff
@@ -225,6 +225,38 @@ jobs:
...
+     working_directory: ~/universal-ctags
+     docker:
+       - image: centos:6
+     steps:
+       - run:
+           name: Rearrange CentOS-Base.repo file
+           command: |
+             sed -i "s|#baseurl=|baseurl=|g" /etc/yum.repos.d/CentOS-Base.repo
+             sed -i "s|mirrorlist=|#mirrorlist=|g" /etc/yum.repos.d/CentOS-Base.repo
+             sed -i "s|http://mirror\.centos\.org/centos/\$releasever|https://vault\.centos\.org/6.10|g" /etc/yum.repos.d/CentOS-Base.repo
```
Comment (leleliu008) [inline circle.yml:238]: `centos:6` and `centos:6.10` are actually the same image.

here we just change `mirror` to `vault` is ok:

```
sed -i 's|mirror|vault|g' /etc/yum.repos.d/CentOS-Base.repo
```

File: circle.yml (hunk 2 above)
Comment (masatake) [reply circle.yml]: Thank you.

Comment (masatake) [reply circle.yml]: I found our test harness needs python 3.5.
misc/units.py uses subprocess.run(), which is introduced in python3.5.
So I will not run the test cases on centos6@circleci.

File: circle.yml (hunk 3 above)
Comment (masatake) [reply circle.yml:240]: Fixed. Thank you.

File: circle.yml (hunk 4 above)
Comment (masatake) [reply circle.yml:238]: Fixed. Thank you.

**Review by leleliu008 — APPROVED** [review]

**Comment by masatake** [comment]
@leleliu008, thank you for your suggestions and review.
Many troubles about the building are gone. The last one is in our queue is only #2791.

**Comment by leleliu008** [comment]
nice!

**Pull request merged**